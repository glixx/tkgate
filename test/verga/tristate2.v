// From http://www.see.ed.ac.uk/~gerard/Teach/Verilog/manual

module triDriver(bus, drive, value);
  inout [3:0] bus;
  input       drive;
  input [3:0] value;

  assign #2 bus = (drive == 1) ? value : 4'bz;

endmodule // triDriver

module myTest;
  wire [3:0] bus;
  reg drive0, drive1, drive2;

  integer    i;

  triDriver mod1 (bus, drive0, i[3:0]);
  triDriver mod2 (bus, drive1, 4'hf);
  triDriver mod3 (bus, drive2, 4'h0);

  initial
    begin
      for (i = 0; i < 12; i = i + 1) begin
        #5 {drive2, drive1, drive0} = i;
        #5 $display ($time,"  %b %b %d", i[2:0], bus, bus);
      end
      $finish;
    end // initial begin

endmodule // myTest
