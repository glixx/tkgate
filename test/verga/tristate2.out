10  0 z z
20  1 1 1
30  10 1111 15
40  11 x11 X
50  100 0 0
60  101 0x0x X
70  110 x x
80  111 x x
90  0 z z
100  1 1001 9
110  10 1111 15
120  11 1x11 X
